pkg/gardenlet/controller/managedseed/managedseed_controller.go

```diff
@@ -31,16 +31,29 @@ func (c *Controller) managedSeedAdd(obj interface{}) {
 		return
 	}
 
-	if managedSeed.Generation != managedSeed.Status.ObservedGeneration || managedSeed.DeletionTimestamp != nil {
+	generationChanged := managedSeed.Generation != managedSeed.Status.ObservedGeneration
+	jitterUpdates := (c.config.Controllers.ManagedSeed.JitterUpdates != nil) && *(c.config.Controllers.ManagedSeed.JitterUpdates)
+
+	// Managed seed with deletion timestamp and newly created managed seed will be enqueued immediately.
+	// Generation is 1 for newly created objects.
+	if managedSeed.DeletionTimestamp != nil || managedSeed.Generation == 1 {
 		c.logger.Debugf("Added ManagedSeed %s without delay to the queue", key)
 		c.managedSeedQueue.Add(key)
+		return
+	}
+
+	if generationChanged {
+		if jitterUpdates {
+			enqueueWithJitterDelay(c, key)
```

Comment:
What about the case:
`c.config.Controllers.ManagedSeed.SyncJitterPeriod` == `0s`
`c.config.Controllers.ManagedSeed.JitterUpdates` == true
Then we would not respect the JitterUpdates field. Do we then want to enqueue with a fallback of `5m` in case of a spec change?

Reply:
We agreed upon specifying in the documentation that the field that the `SyncJitterPeriod` determines the period duration also for `JitterUpdates`

Reply:
Basically the case that somebody disables SyncJitterPeriod but enables JitterUpdates is currently not supported. We can change that if we see there is a need for that which we currently don't predict to happen.